File: 19_09_2022/2325._Decode_the_Message.py
class Solution:
    def decodeMessage(self, key: str, message: str) -> str:
        d={}
        ans=""
        alp='a'
        for i in key:
            if(i!=" " and i not in d):
                d[i]=alp
                alp=chr(ord(alp)+1)
        print(d)
        for i in message:
            if(i!=" "):
                ans+=d[i]
            else:
                ans+=" "
        return ans
                
            
        

File: 19_09_2022/2325._Decode_the_Message.cpp
class Solution {
public:
    string decodeMessage(string key, string message) {
        string ans="";
        map<char,int> d;
        char alp='a';
        for(auto x:key)
        {
            if(x!=' ')
            {
                if(d[x]==0)
                {
                    d[x]=alp;
                    alp++;
                }
            }
        }
        
        for(auto xx:message)
        {
            if(xx!=' ')
            {
                ans+=d[xx];
            }
            else
            {
                ans+=" ";
            }
        }
        return ans;
    }
};